docs/en/resources/tools/firestore/firestore-update-document.md

```diff
@@ -0,0 +1,321 @@
+}
+```
+
+### Update with All Data Types
+
+```json
+{
+  "documentPath": "test-documents/doc789",
+  "documentData": {
+    "stringField": {
+      "stringValue": "Updated string"
+    },
+    "integerField": {
+      "integerValue": "999"
+    },
+    "doubleField": {
+      "doubleValue": 2.71828
+    },
+    "booleanField": {
+      "booleanValue": false
+    },
+    "nullField": {
+      "nullValue": null
+    },
+    "timestampField": {
+      "timestampValue": "2025-01-15T16:45:00Z"
+    },
+    "geoPointField": {
+      "geoPointValue": {
+        "latitude": 51.5074,
+        "longitude": -0.1278
+      }
+    },
+    "bytesField": {
+      "bytesValue": "VXBkYXRlZCBkYXRh"
+    },
+    "arrayField": {
+      "arrayValue": {
+        "values": [
+          {
+            "stringValue": "updated1"
+          },
+          {
+            "integerValue": "200"
+          },
+          {
+            "booleanValue": true
+          }
+        ]
+      }
+    },
+    "mapField": {
+      "mapValue": {
+        "fields": {
+          "nestedString": {
+            "stringValue": "updated nested value"
+          },
+          "nestedNumber": {
+            "doubleValue": 88.88
+          }
+        }
+      }
+    },
+    "referenceField": {
+      "referenceValue": "users/updatedUser"
+    }
+  },
+  "returnData": true
+}
+```
+
+## Authentication
+
+The tool can be configured to require authentication:
+
+```yaml
+tools:
+  secure-update-doc:
+    kind: firestore-update-document
+    source: prod-firestore
+    description: Update documents with authentication required
+    authRequired:
+      - google-oauth
+      - api-key
+```
+## Error Handling
+Common errors include:
+- Document not found (when using update with a non-existent document)
+- Invalid document path
+- Missing or invalid document data
+- Permission denied (if Firestore security rules block the operation)
+- Invalid data type conversions
+- Field not found in document data when using update mask
+## Best Practices
+1. **Use update masks for precision**: When you only need to update specific fields, use the `updateMask` parameter to avoid unintended changes
+2. **Always use typed values**: Every field must be wrapped with its appropriate type indicator (e.g., `{"stringValue": "text"}`)
+3. **Integer values can be strings**: The tool accepts integer values as strings (e.g., `{"integerValue": "1500"}`)
+4. **Use returnData sparingly**: Only set to true when you need to verify the exact data after the update
+5. **Validate data before sending**: Ensure your data matches Firestore's native JSON format
+6. **Handle timestamps properly**: Use RFC3339 format for timestamp strings
+7. **Base64 encode binary data**: Binary data must be base64 encoded in the `bytesValue` field
+8. **Consider security rules**: Ensure your Firestore security rules allow document updates
+9. **Use deleteValue for field removal**: When using update mask, set `{"deleteValue": true}` to remove fields
```

Comment:
I'm a little hesitant to add deleteValue because it is not Firestore specific ie not the firestore native json syntax. Firestore uses masks and request bodies to delete values.